### PR TITLE
Doors usage improvements

### PR DESCRIPTION
**Pull request by akortunov**
1. AI can use keys to open doors now.
2. A door opening sound will be played when AI opens a door.
3. Search for key starts only if this door can be unlocked by key and ends if the key found.

Note: I took search from door.cpp. Maybe we should create a separate function for this search?

### CODE REVIEW

**Comment by kcat** [comment]
This sounds like something that could easily break the game, with NPCs opening doors that the player isn't supposed to be able to get through.

**Comment by akortunov** [comment]
Actually, AI can use keys in vanilla Morrowind.

File: apps/openmw/mwmechanics/aipackage.cpp [hunk 1]
```diff
@@ -179,10 +180,33 @@ void MWMechanics::AiPackage::evadeObstacles(const MWWorld::Ptr& actor, float dur
     if (door != MWWorld::Ptr())
     {
         // note: AiWander currently does not open doors
-        if (getTypeId() != TypeIdWander && !door.getCellRef().getTeleport() && door.getCellRef().getTrap().empty()
-                && door.getCellRef().getLockLevel() <= 0 && door.getClass().getDoorState(door) == 0)
+        if (getTypeId() != TypeIdWander && !door.getCellRef().getTeleport() && door.getClass().getDoorState(door) == 0)
         {
-            MWBase::Environment::get().getWorld()->activateDoor(door, 1);
+            if ((door.getCellRef().getTrap().empty() && door.getCellRef().getLockLevel() <= 0 ))
+                MWBase::Environment::get().getWorld()->activate(door, actor);
```
Comment (scrawl) [inline apps/openmw/mwmechanics/aipackage.cpp]: Looks like you call activate() twice for a door that needs a key but is already unlocked?

Comment (akortunov) [inline apps/openmw/mwmechanics/aipackage.cpp]: Yes, you are right. I missed a return statement. Sorry.

**Pull request merged**